### PR TITLE
DDCE-3737: CSP filtering on PBIK

### PR DESCRIPTION
**Pull request by rreddy80**
# DDCE-3737

ensure the PBIK service is protected against cross-site scripting attacks (XSS)

## Checklist PR Raiser
- [x]  I've updated the README if there are functionality changes
- [x]  I've executed the project script to check Dependencies and update them if they're non-breaking changes
- [x]  I've executed Scalastyle and fixed any warnings that can be resolved quickly
- [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
- [x]  I've included appropriate tests with any code I've added (Unit, Integration), and it meets Coverage
- [x]  I've squashed my commits - including the JIRA issue number in the commit message
- [ ]  I've contacted a BA to create a task if any of these actions would result in a larger piece of work

### CODE REVIEW

File: app/views/ErrorPage.scala.html [hunk 1]
```diff
@@ -65,13 +67,20 @@ <h1 id="title" class="govuk-heading-xl">
...
         @code match {
-            case 63082 => {<a class="govuk-link" id="link-try-again" href='javascript: window.history.go(-1)'>@messages("Service.tryagain")</a>}
-            case 63090 | 63092  => {<a class="govuk-link" id="link-add-another" href='javascript: window.history.go(-1)'>@messages("Service.excludeanotheremployee")</a>}
+            case 63082 => {
+                <a class="govuk-link" id="link-try-again">@messages("Service.tryagain")</a>
+                <script @{CSPNonce.attr}>document.getElementById("link-try-again").addEventListener("click", () => window.history.go(-1));</script>
+            }
+            case 63090 | 63092  => {
+                <a class="govuk-link" id="link-add-another">@messages("Service.excludeanotheremployee")</a>
+                <script @{CSPNonce.attr}>document.getElementById("link-add-another").addEventListener("click", () => window.history.go(-1));</script>
```
Comment (lmugnai) [inline app/views/ErrorPage.scala.html:82]: These 2 scripts are basically doing the same thing. Is there anyway of combining them? Also, would it not be best to keep all Script tags together towards the top of the body?

File: app/views/ErrorPage.scala.html [hunk 2]
```diff
@@ -52,10 +53,11 @@ <h1 id="title" class="govuk-heading-xl">
      @isCurrentTaxYear match {
             case _ if(code == 63091 || code == 65127 || code == 63085) => {
                 @Html(messages("ServiceMessage." + (code)))
-            }
+                <script @{CSPNonce.attr}>document.querySelectorAll("#back-link-63091, #link-try-again").forEach(elem => elem.addEventListener("click", () => window.history.go(-1)));</script>
```
Comment (lmugnai) [inline app/views/ErrorPage.scala.html:56]: was there a #back-link-63091 on the page before? Is there one now? I can't see it. Also, there's a script tag for #link-try-again below so why do we need it here too?

File: app/views/registration/ConfirmUpdateNextTaxYear.scala.html [hunk 3]
```diff
@@ -41,7 +42,7 @@ <h1 id="title" class="govuk-heading-xl">
     </h1>
 
     <div class="data">
-        @formWithCSRF(action = controllers.registration.routes.ManageRegistrationController.addNextYearRegisteredBenefitTypes, Symbol("onsubmit") -> "gaEventConfirmAddBenefits()") {
+        @formWithCSRF(action = controllers.registration.routes.ManageRegistrationController.addNextYearRegisteredBenefitTypes) {
```
Comment (lmugnai) [inline app/views/registration/ConfirmUpdateNextTaxYear.scala.html:45]: I think I need this explained why it is removed :)

File: app/views/ErrorPage.scala.html (hunk 2 above)
Comment (lmugnai) [reply app/views/ErrorPage.scala.html:56]: oh, i see #back-link-63091 href is in the messages file.

Comment (rreddy80) [reply app/views/ErrorPage.scala.html:56]: yes some messages have html tags in them and is one of them. There're a few I don't want to change and leave them as is, at the same time apply CSP. so given the html tag an id attribute and if that error occurs, dynamically a script is added so the page doesn't violate the CSP

File: app/views/ErrorPage.scala.html (hunk 1 above)
Comment (rreddy80) [reply app/views/ErrorPage.scala.html:82]: true they do the same. However they're not going to be added at the same time. we might hit either of those errors and in such an instance we have a script to apply CSP on it. (These are some of the errors where error message has javascript code in it.) Anyway, if we add the tag without the anchor tag rendered on the page first, script tag will throw an error. so had to implement it like that

File: app/views/registration/ConfirmUpdateNextTaxYear.scala.html (hunk 3 above)
Comment (rreddy80) [reply app/views/registration/ConfirmUpdateNextTaxYear.scala.html:45]: the scripts to add the google analytics events were removed as part of a ticket 2yrs ago (May/2021) after that bit of clean-up was done however the cleanup was not done properly. And this is the remnants of that clean-up. We should have been hitting a console error in production as well cause of this, cause we're trying to call a function that's not available any more

**Review by lmugnai — APPROVED** [review]

**Pull request merged**